daemon/src/wallet.rs

```diff
@@ -0,0 +1,27 @@
+use crate::model::WalletInfo;
+use anyhow::Result;
+use bdk::bitcoin::Amount;
+use bdk::blockchain::{Blockchain, NoopProgress};
+use bdk::database::BatchDatabase;
+use bdk::wallet::AddressIndex;
+use std::time::SystemTime;
+
+pub fn wallet_info<B, D>(wallet: &bdk::Wallet<B, D>) -> Result<WalletInfo>
```

Comment:
Call this function `sync`, that describes better what actually happens in here. With the rebase on top of @klochowicz's work, perhaps make it a function on `Wallet` right away.